polkadot/node/subsystem-bench/src/lib/network.rs

```diff
@@ -728,8 +746,41 @@ pub struct NetworkEmulatorHandle {
 }
 
 impl NetworkEmulatorHandle {
+	pub fn generate_statement_distribution_peer_view_change(&self, view: View) -> Vec<AllMessages> {
+		self.peers
+			.iter()
+			.filter(|peer| peer.is_connected())
+			.map(|peer| {
+				AllMessages::StatementDistribution(
+					StatementDistributionMessage::NetworkBridgeUpdate(
+						NetworkBridgeEvent::PeerViewChange(peer.peer_id(), view.clone()),
+					),
+				)
+			})
+			.collect_vec()
+	}
+	/// Generates peer_connected messages for all peers in `test_authorities`
+	pub fn generate_statement_distribution_peer_connected(&self) -> Vec<AllMessages> {
+		self.peers
+			.iter()
+			.filter(|peer| peer.is_connected())
+			.map(|peer| {
+				let network = NetworkBridgeEvent::PeerConnected(
+					peer.handle().peer_id,
+					ObservedRole::Authority,
+					ValidationVersion::V3.into(),
+					Some(vec![peer.authority_id()].into_iter().collect()),
+				);
+
+				AllMessages::StatementDistribution(
+					StatementDistributionMessage::NetworkBridgeUpdate(network),
+				)
+			})
+			.collect_vec()
+	}
+
 	/// Generates peer_connected messages for all peers in `test_authorities`
-	pub fn generate_peer_connected(&self) -> Vec<AllMessages> {
+	pub fn generate_approval_distribution_peer_connected(&self) -> Vec<AllMessages> {
```

Comment:
Nit: To not repeat these 2 functions you can name it something like

generate_subsystem_peer_connected(&self, lambda_to_transform_peer_connected_subsystem_specific_message)

and then call the lambda function in the map.